### ISSUE TITLE
NPE in Websphere AS 8.0.0.9 when using enableRestorableView

### ISSUE THREAD

**Issue by volkovd** [omnifaces/omnifaces#73]
I am getting NullPointerException when ViewState cannot be found and is restored by RestorableViewHandler in following scenario: 

1. Load page1 (with o:enableRestorableView) 
2. Load page2 in new browser window and execute some dummy action there to provoke ViewExpiredException
3. Go back to the page1 and execute some action there. - Action is executed __without__ any exceptions
4. Execute same action again - following exception is thrown:  
```
java.lang.NullPointerException
	at java.lang.Class.isAssignableFrom(Native Method)
	at javax.faces.component.UIComponent$EventListenerWrapper.isListenerForSource(UIComponent.java:1111)
	at org.apache.myfaces.application.ApplicationImpl._traverseListenerList(ApplicationImpl.java:2121)
	at org.apache.myfaces.application.ApplicationImpl.publishEvent(ApplicationImpl.java:551)
	at javax.faces.application.ApplicationWrapper.publishEvent(ApplicationWrapper.java:329)
	at org.apache.webbeans.jsf.OwbApplication.publishEvent(OwbApplication.java:474)
	at javax.faces.application.ApplicationWrapper.publishEvent(ApplicationWrapper.java:329)
	at javax.faces.application.ApplicationWrapper.publishEvent(ApplicationWrapper.java:329)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:479)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage._publishPostBuildComponentTreeOnRestoreViewEvent(FaceletViewDeclarationLanguage.java:500)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage.buildView(FaceletViewDeclarationLanguage.java:375)
	at org.apache.myfaces.lifecycle.RenderResponseExecutor.execute(RenderResponseExecutor.java:66)
	at org.apache.myfaces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:240)
	at org.apache.myfaces.extensions.cdi.jsf2.impl.listener.phase.CodiLifecycleWrapper.render(CodiLifecycleWrapper.java:126)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:191)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:779)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:478)
	at com.ibm.ws.webcontainer.servlet.ServletWrapperImpl.handleRequest(ServletWrapperImpl.java:178)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:136)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:97)
	at org.apache.myfaces.webapp.filter.ExtensionsFilter.doFilter(ExtensionsFilter.java:357)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at net.bull.javamelody.MonitoringFilter.doFilter(MonitoringFilter.java:163)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at net.bull.javamelody.MonitoringFilter.doFilter(MonitoringFilter.java:206)
	at net.bull.javamelody.MonitoringFilter.doFilter(MonitoringFilter.java:179)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at de.eis.fba2.jsf.CharsetFilter.doFilter(CharsetFilter.java:23)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at de.eis.fba2.cache.NoCacheFilter.doFilter(NoCacheFilter.java:26)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at de.eis.fba2.cache.CacheFilter.doFilter(CacheFilter.java:22)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at de.eis.fba2.security.NoAccessFilter.doFilter(NoAccessFilter.java:21)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at de.eis.fba2.logging.MDCServletFilter.doFilter(MDCServletFilter.java:49)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:195)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:960)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1064)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:87)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:914)
	at com.ibm.ws.webcontainer.WSWebContainer.handleRequest(WSWebContainer.java:1662)
	at com.ibm.ws.webcontainer.channel.WCChannelLink.ready(WCChannelLink.java:200)
	at com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:453)
	at com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.handleNewRequest(HttpInboundLink.java:515)
	at com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.processRequest(HttpInboundLink.java:306)
	at com.ibm.ws.http.channel.inbound.impl.HttpICLReadCallback.complete(HttpICLReadCallback.java:84)
	at com.ibm.ws.tcp.channel.impl.AioReadCompletionListener.futureCompleted(AioReadCompletionListener.java:175)
	at com.ibm.io.async.AbstractAsyncFuture.invokeCallback(AbstractAsyncFuture.java:217)
	at com.ibm.io.async.AsyncChannelFuture.fireCompletionActions(AsyncChannelFuture.java:161)
	at com.ibm.io.async.AsyncFuture.completed(AsyncFuture.java:138)
	at com.ibm.io.async.ResultHandler.complete(ResultHandler.java:204)
	at com.ibm.io.async.ResultHandler.runEventProcessingLoop(ResultHandler.java:775)
	at com.ibm.io.async.ResultHandler$2.run(ResultHandler.java:905)
	at com.ibm.ws.util.ThreadPool$Worker.run(ThreadPool.java:1702)
 

```

**Comment by BalusC** [comment]
Which OmniFaces version? Which MyFaces version? Tried latest MyFaces?

**Comment by arjantijms** [comment]
>Which MyFaces version?

As OP mentioned Websphere AS 8.0.0.9 it's probably the oldest most ancient MyFaces version that IBM by some miracle has been able to get certified. If memory serves me correctly it was a modified Myfaces 2.0.4 from early 2011.

>Tried latest MyFaces?

You can't really update MyFaces in WebSphere since they use a modified version. My guess is they went a bit too far with their modifications and can't update it themselves either, since all later WebSphere and Liberty products (e.g. 8.5.x) have the same ancient and bug ridden MyFaces version.

**Comment by volkovd** [comment]
Absolutely correct. It is not possible to update MyFaces Version in WebSphere.

**Comment by BalusC** [comment]
We recently fixed a bunch of potential serialization errors (and inherently also view state related) for OmniFaces 2.1. Do you see chance to try 2.1 snapshot and see if it fixes your particular problem?

**Comment by volkovd** [comment]
I would like to, but OmniFaces 2.x requires JSF 2.2, as far as I know. JSF 2.2 is not available in Websphere.

**Comment by pthorson** [comment]
It is not an option for us so we have not tried it, but note that the Websphere documentation does provide instructions for using a different JSF version (http://www-01.ibm.com/support/knowledgecenter/SSEQTP_8.5.5/com.ibm.websphere.base.doc/ae/tweb_jsf.html?cp=SSEQTP_8.5.5%2F1-12-2-33-0-2-3):
**Be Aware**: If you want to use a third-party JSF implementation that is not shipped with the product, then:
1. Add the third-party listener to the required web.xml file.
2. Add the third-party JSF implementation Java archive (JAR) files and its required dependant JAR files to the application as an isolated shared library and associate it with your application:
  * Move the JSF JAR files and all 3rd party libraries referenced by the JSF JAR files and UI component plugins (for example, PrimeFaces) to a directory outside of the application.
  * Create the isolated shared library. See topic Creating shared libraries for information about how to create the shared library.
  * Ensure the classloader is set to PARENT_FIRST, the default value. To view the current configuration in the administrative console panel, click Applications > Application Types > WebSphere enterprise applications > application_name > Class loading and update detection.
  * If you are using a third-party *MyFaces* implementation, set the JSF implementation to Sun RI. To change this configuration in the administrative console panel, click Applications > Application Types > WebSphere enterprise applications > application_name > JSP and JSF options. Under JSF implementation, select Sun RI from the drop-down menu.
  * If you are using a *different* third-party JSF implementation, keep the JSF implementation set to the default value.

Restart the application server.

**Comment by BalusC** [comment]
> I would like to, but OmniFaces 2.x requires JSF 2.2, as far as I know. JSF 2.2 is not available in Websphere.

Those fixes are also present in 1.11. Can you give it a try if possible?

**Comment by arjantijms** [comment]
Ps

JSF 2.2 (MyFaces) is now available in Liberty 8.5.5.6.

**Comment by BalusC** [comment]
Closing off because of no respons and problem is theoretically already fixed.

If problem still persists, please open a new issue.
